### DISCUSSION TITLE
LM studio connection failed

### DISCUSSION THREAD

**Discussion by TheophileCAE** [open-webui/open-webui#3573] [General]
# Bug Report

## Description
**Bug Summary:**
When trying to retrieve models from LM studio for inference I get "OpenAI: Network Problem" as red error on top of the screen.

**Steps to Reproduce:**
- I installed Open WebUI docker with command : ```docker run -d -p 3000:8080 --gpus=all -v ollama:/root/.ollama -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:ollama```
- I started ollama server and lm studio server
- The ollama server was automatically connected and worked
- In admin setting i set the openAI based url as the lm studio server url `http://localhost:5001/v1` and the key as `lm-studio'
- I also tried no key and `none` key
- I saved the setting and get this error.

**Expected Behavior:**
It is expected to have the list of lm studio downloaded models and a connection request in lm studio log server

**Actual Behavior:**
![Capture d’écran 2024-07-01 094533](https://github.com/open-webui/open-webui/assets/169677785/c8d5f35e-baab-4d3a-829f-8222835ad6ec)

## Environment

- **Open WebUI Version:** v0.3.6
- **Ollama (if applicable):** 0.1.44

- **Operating System:** Windows 11 Pro
- **Browser (if applicable):**  Firefox 127.0.2 (64-bit), Opera

## Reproduction Details

**Confirmation:**

- [x ] I have read and followed all the instructions provided in the README.md.
- [x ] I am on the latest version of both Open WebUI and Ollama.
- [x ] I have included the browser console logs.
- [x ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
![image](https://github.com/open-webui/open-webui/assets/169677785/1b49460a-94f5-4ebb-853a-532d5f9b8038)
![image](https://github.com/open-webui/open-webui/assets/169677785/348f7bf5-f6d8-45b1-92e2-2415fa3e2840)
![image](https://github.com/open-webui/open-webui/assets/169677785/b779c2c2-3c3b-4176-bc8e-5f065ec78489)

**Docker Container Logs:**
![image](https://github.com/open-webui/open-webui/assets/169677785/1f58d598-920a-4ba1-8cc6-ff60b995dc2e)

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

```docker run -d -p 3000:8080 --gpus=all -v ollama:/root/.ollama -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:ollama```

I also tried installing the docker for OpenAI only and I get the same result.
```docker run -d -p 3000:8080 -e OPENAI_API_KEY=your_secret_key -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main```

## Additional Information

I tried using get request on post man to make sure the issue wasn't on lm studio server and it worked :
![image](https://github.com/open-webui/open-webui/assets/169677785/d8a5d577-afed-4ebc-ac96-b53a467a7a9e)

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by asmj1108** [comment]
Within a Docker container, `localhost` refers to the container itself, not your actual host system.

As referenced [here](https://www.howtogeek.com/devops/how-to-connect-to-localhost-within-a-docker-container/): 
> `localhost` and `127.0.0.1`: These addresses resolve to the container's network,
> `host.docker.internal`: This address allows the container to communicate with the host machine.

So the OpenAI base URL for the LM Studio server should be `http://host.docker.internal:1234/v1` on Windows and macOS.

On Linux, however, Docker does not automatically support the `host.docker.internal` alias. To use this alias, the container must be started with the `--add-host host.docker.internal:host-gateway` option.

**Reply by f4lc0n90** [reply]
Thank you very much, it works perfectly
